src/index.tsx

```diff
@@ -44,6 +38,23 @@ export default class ReactToPrint extends React.Component<IReactToPrintProps> {
                 onAfterPrint();
             }
         }, 500);
+    }
+
+    triggerPrint = (target) => {
+        const { onBeforePrint, onAfterPrint } = this.props;
+
+        if (onBeforePrint) {
+            const onBeforePrintOutput = onBeforePrint();
+            if (onBeforePrintOutput instanceof Promise) {
+                onBeforePrintOutput.then(r => {
+                    this.startPrint(target, onAfterPrint);
+                }).catch(e => {
```

Comment:
I don't think we should `catch` this failure. If we do this then if the user chains a `.catch` it won't get fired.

Reply:
Done.